src/backend/distributed/operations/create_shards.c

```diff
@@ -362,6 +376,65 @@ CreateReferenceTableShard(Oid distributedTableId)
 }
 
 
+/*
+ * CreateNullKeyShard creates a single shard for the given
+ * distributedTableId. The created shard does not have min/max values.
+ * Unlike CreateReferenceTableShard, the shard is _not_ replicated to
+ * all nodes but would have a single placement like Citus local tables.
+ * However, this placement doesn't necessarily need to be placed on
+ * coordinator.
+ */
+void
+CreateNullKeyShard(Oid relationId, bool useExclusiveConnection)
+{
+	EnsureTableOwner(relationId);
+
+	/* we plan to add shards: get an exclusive lock on relation oid */
+	LockRelationOid(relationId, ExclusiveLock);
+
+	/* don't allow concurrent node list changes that require an exclusive lock */
+	LockRelationOid(DistNodeRelationId(), RowShareLock);
+
+	/* load and sort the worker node list for deterministic placement */
+	List *workerNodeList = DistributedTablePlacementNodeList(NoLock);
+	workerNodeList = SortList(workerNodeList, CompareWorkerNodes);
+
+	int32 workerNodeCount = list_length(workerNodeList);
+	if (workerNodeCount == 0)
+	{
+		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						errmsg("couldn't find any worker nodes"),
+						errhint("Add more worker nodes")));
+	}
+
+	static int roundRobinNodeIdx = -1;
+	roundRobinNodeIdx++;
```

Comment:
This seems local to the session. Is there a reliable way to make it round robin across sessions? (e.g. using shard ID)